Sources/Sentry/SentrySpan.m

```diff
@@ -84,10 +96,46 @@ - (instancetype)initWithContext:(SentrySpanContext *)context
         _spanId = context.spanId;
         _sampled = context.sampled;
         _origin = context.origin;
+
+#if SENTRY_TARGET_PROFILING_SUPPORTED
+        if (SentrySDK.options.enableContinuousProfiling) {
+            _profileSessionIDs = [NSMutableSet<NSString *> set];
+            if (SentryContinuousProfiler.isCurrentlyProfiling) {
+                [_profileSessionIDs
+                    addObject:SentryContinuousProfiler.currentProfilerID.sentryIdString];
+            } else {
```

Comment:
`m`: Shouldn't we also add the observer if we are already profiling? Can it happen that a span has multiple profileSessionIDs?

The test case for this would be
```
    +---------span-----------+
 +--profile1--+    +--profile2--+
```

Reply:
Yep you are right for the original implementation, but this is now obviated.